server/routes/ad.ts

```diff
@@ -260,19 +260,29 @@ export default class AdService {
 
       // Get real-time and historical task info to populate the
       // task and job-related fields
-      const realtimeTaskResponse: any = await this.client
-        .asScoped(request)
-        .callAsCurrentUser('ad.searchTasks', {
-          body: getLatestTaskForDetectorQuery(detectorId, true),
-        });
-      const historicalTaskResponse: any = await this.client
-        .asScoped(request)
-        .callAsCurrentUser('ad.searchTasks', {
-          body: getLatestTaskForDetectorQuery(detectorId, false),
-        });
+      // We wrap these calls in a try/catch, and suppress any index_not_found_exceptions
+      // which can occur if no detector jobs have been ran on a new cluster.
+      let realtimeTasksResponse = {} as any;
+      let historicalTasksResponse = {} as any;
+      try {
+        realtimeTasksResponse = await this.client
+          .asScoped(request)
+          .callAsCurrentUser('ad.searchTasks', {
+            body: getLatestTaskForDetectorQuery(detectorId, true),
+          });
+        historicalTasksResponse = await this.client
+          .asScoped(request)
+          .callAsCurrentUser('ad.searchTasks', {
+            body: getLatestTaskForDetectorQuery(detectorId, false),
+          });
+      } catch (err) {
+        if (!isIndexNotFoundError(err)) {
```

Comment:
Are there any critical indices that could be not found other then `.opendistro-anomaly-detection-state` where you would want indexNotFoundError to be displayed or anyway giving an error on getting detector list for an issue with search always a 'buggy' output.

Reply:
Not for this specific call. If there was other critical indices missing they would already get caught and propagated as an error in the previous calls, such as fetching the initial detector info.

Since this call is just for searching the tasks which is querying the `.opendistro-anomaly-detection-state` index, it should be ok. Prefer to not hardcode many system indices on the frontend, since I would consider it more to be a backend implementation detail.

Reply:
Makes sense, sounds good to me.